keps/sig-api-machinery/20190802-dynamic-coordinated-storage-version.md

```diff
@@ -172,6 +172,34 @@ During bootstrap, for each resource, the API server
   to avoid conflicting with other API servers.
 * installs the resource handler.
 
+The above mentioned process requires an API server to update the storageVersion
+before accepting API requests. If we don't enforce this order, data encoded in
+an unexpected version can sneak into etcd. For example, an API server persists a
+write request encoded in an obsoleted version, then it crashes before it can
+update the storageVersion. The storage migrator has no way to detect this write.
+
+For the cmd/kube-apiserver binary, we plan to enforce this order by adding a new
+filter to the [handler chain][]. Before kube-aggregator, kube-apiserver, and
+apiextension-apiserver have registered the storage version of the built-in
+resources they host, this filter only allows the following requests:
```

Comment:
Can we add aggregated API requests to this list?  Requests to create resources in places like service catalog and openshift could logically proceed.

Reply:
Added. Also excluded API defined by CRD. This filter only blocks built-in resources.